app/bundles/CoreBundle/Form/Type/SortableValueLabelListType.php

```diff
@@ -29,11 +32,27 @@ public function buildForm(FormBuilderInterface $builder, array $options): void
             'value',
             TextType::class,
             [
-                'label'          => 'mautic.core.value',
+                'label'          => 'mautic.core.value_autogenerated',
                 'error_bubbling' => true,
+                'required'       => false,
                 'attr'           => ['class' => 'form-control'],
             ]
         );
+
+        $builder->addEventListener(
+            FormEvents::PRE_SUBMIT,
+            function (FormEvent $event): void {
+                $data = $event->getData();
+
+                if (is_array($data)
+                    && empty($data['value'])
+                    && !empty($data['label'])
+                ) {
+                    $data['value'] = strtolower(preg_replace('/_+/', '_', trim(InputHelper::alphanum(InputHelper::transliterate((string) $data['label']), false, '_'), '_')));
```

Comment:
This is hard to read and debug. Consider extracting to a private method:

  private function generateSlugFromLabel(string $label): string
  {
      $transliterated = InputHelper::transliterate($label);
      $alphanumeric = InputHelper::alphanum($transliterated, false, '_');
      $trimmed = trim($alphanumeric, '_');
      $normalized = preg_replace('/_+/', '_', $trimmed);
      return strtolower($normalized);
  }

Or use only already created and tested method, even If we do not cover all situations.